src/components/AgenciesPage/components/AgenciesList.tsx

```diff
@@ -1,14 +1,32 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { Grid } from '@mui/material';
+import { useDispatch } from 'react-redux';
+import { Dispatch } from '@reduxjs/toolkit';
 import { AgenciesItem } from './components';
-import { agenciesData } from '../mocks/agenciesData';
-
-export const AgenciesList = () => (
-  <Grid container>
-    {agenciesData.map(agency => (
-      <Grid item key={agency.id} xs={12} sm={6} md={4} lg={3}>
-        <AgenciesItem agency={agency} />
-      </Grid>
-    ))}
-  </Grid>
-);
+import { selectAgencies } from '../selectors/selectors';
+import { getAgenciesData } from '../services/getAgenciesData';
+import './AgenciesList.scss';
+
+export const AgenciesList = () => {
+  const dispatch = useDispatch<Dispatch>();
+  const { agencies } = selectAgencies();
+
+  useEffect(() => {
+    getAgenciesData()(dispatch);
+  }, []);
+
+  if (!agencies.length)
+    return (
+      <h2 className="agencies-list-empty">No agencies for now. Try later.</h2>
+    );
+
+  return (
+    <Grid container>
+      {agencies.map(agency => (
+        <Grid item key={agency.id} xs={12} sm={6} md={4} lg={3}>
+          <AgenciesItem agency={agency} />
+        </Grid>
+      ))}
+    </Grid>
```

Comment:
don't see the logic there whether data is loading

Reply:
do you mean to add some kinder of loader?

Reply:
you set loading field to the state, but don't use it in the component